src/parser/SGTranspileState.ts

```diff
@@ -0,0 +1,75 @@
+import type { Range } from 'vscode-languageserver';
+import { SourceNode } from 'source-map';
+import type { SGAttribute, SGToken } from './SGTypes';
+
+export class SGTranspileState {
```

Comment:
I don't really see a point in creating a separate TranspileState class just for SceneGraph. The unique features of this class could easily be moved into a utility class, and then we can share `TranspileState` between Brs and XML files.

Reply:
I tried to use `TranspileState` but it was way too specific to Brs. It still feels convenient as you have less properties to pass through to all the transpile methods.